### ISSUE TITLE
[BUG] Screenshot width height is not matching device Viewport sizes

### ISSUE THREAD

**Issue by Jamesj001** [microsoft/playwright#23460]
### System info
- Playwright Version: [1.30.0]
- Operating System: [ macOS 13.2, etc.]
- Browser: [Mobile Chromium]
- Other info:

### Source code

- [ ] I provided exact source code that allows reproducing the issue locally.

or

**Config file**

```js
// playwright.config.ts
import { defineConfig, devices } from '@playwright/test';

export default defineConfig({
  projects: [
    {
      name: 'desktop-chromium',
      use: { ...devices['Desktop Chrome'] },
    },
    {
      name: 'mobile-chromium',
      use: { ......devices['Pixel 5'] },
    },
  ],
});
```
The snapshot taken has the width and height as W: 980 H: 1812 for mobile chromium device but the viewport size for mobile chromium i.e. Pixel 5 we see as width : 393, height: 727 --->

```
"Pixel 5": {
    "userAgent": "Mozilla/5.0 (Linux; Android 11; Pixel 5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/115.0.5790.13 Mobile Safari/537.36",
    "screen": {
      "width": 393,
      "height": 851
    },
    "viewport": {
      "width": 393,
      "height": 727
    },
    "deviceScaleFactor": 2.75,
    "isMobile": true,
    "hasTouch": true,
    "defaultBrowserType": "chromium"
  }, 

```
Am i missing something here ?

**Comment by mxschmitt** [comment]
For me it was working correctly. Do you mind sharing a minimal reproducible? And which OS you use? Also your config wasn't working, so I guess it was not your actual one.

<img width="383" alt="image" src="https://github.com/microsoft/playwright/assets/17984549/15820df2-2efc-4429-9798-53e4dc2f716a">

**Comment by Jamesj001** [comment]
Yeah @mxschmitt I used this : `await expect(page).toHaveScreenshot() `

**Comment by mxschmitt** [comment]
Can you share a repro which I can try to run locally?

**Comment by aslushnikov** [comment]
Unfortunately, this is not actionable without a reproduction, so I'll have to close this.

Feel free to file another issue with a reproduction if you think we can be helpful!
